### PR TITLE
Add LuGilaSilverRiftGuardian (dm06)

### PR DESCRIPTION
**Pull request by DanieloV**
+ Add Lu Gila, Silver Rift Guardian (dm06)

### CODE REVIEW

File: game/cards/dm06/guardian.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	c.Name = "Lu Gila, Silver Rift Guardian"
+	c.Power = 4000
+	c.Civ = civ.Light
+	c.Family = []string{family.Guardian}
+	c.ManaCost = 5
+	c.ManaRequirement = []string{civ.Light}
+
+	c.Use(fx.Creature, fx.Blocker, fx.CantAttackPlayers, func(card *match.Card, ctx *match.Context) {
+
+		if card.Zone != match.BATTLEZONE && card.Zone != match.HIDDENZONE {
```
Comment (sindreslungaard) [inline game/cards/dm06/guardian.go:22]: I don't think the HIDDENZONE check is needed here (?) If you put another evolution card on this one (stack of 3) only the top card's effects should apply as far as I know

File: game/cards/dm06/guardian.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+			return
+		}
+
+		if event, ok := ctx.Event.(*match.CardMoved); ok {
+			if event.To != match.BATTLEZONE {
+				return
+			}
+
+			playedCard, ok := ctx.Match.Player1.Player.GetCard(event.CardID, match.BATTLEZONE)
```
Comment (sindreslungaard) [inline game/cards/dm06/guardian.go:31]: "ok" should probably be called "err" as per Go convention

File: game/cards/dm06/guardian.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+			if !playedCard.HasCondition(cnd.Evolution) {
+				return
+			}
+
+			// If the card is hidden, make sure it was the one that just evolved
+			at := playedCard.Attachments()
+			if card.Zone == match.HIDDENZONE && at[len(at)-1] != card {
+				return
+			}
```
Comment (sindreslungaard) [inline game/cards/dm06/guardian.go:47]: What is this for?

If it is in fact needed then we should probably add a check to make sure we don't panic the program by indexing into `at[-1]` if in the future there's an evolution card that can be played without any attachments for some reason

Comment (DanieloV) [reply game/cards/dm06/guardian.go:47]: According to the rulings (https://duelmasters.fandom.com/wiki/Lu_Gila,_Silver_Rift_Guardian/Rulings) if you evolve Lu Gila, it's evolution also becomes tapped. But good point with the indexing will add a check for that and improve the explanation.

File: game/cards/dm06/guardian.go (hunk 1 above)
Comment (DanieloV) [reply game/cards/dm06/guardian.go:22]: This is needed for the case where Lu GIla was the card that just evolved. According to the rulings (https://duelmasters.fandom.com/wiki/Lu_Gila,_Silver_Rift_Guardian/Rulings) if you evolve Lu Gila, it's evolution also becomes tapped.

**Comment by DanieloV** [comment]
@sindreslungaard applied changes and answered the comments

**Pull request merged**